docs/posts/contribute.md

```diff
@@ -13,12 +14,17 @@ If you have any questions about any of this stuff, just ask! :)
 We'd love to hear from you! Please submit any bug reports and feature requests through [Github issues](https://github.com/chanzuckerberg/cellxgene/issues).
 
 # Direct contributions
+
 ## Getting started
 
 If you are interested in working on `cellxgene` development, you'll need to use git to make a copy of the [project repository](https://www.youtube.com/watch?v=A-4WltCTVms&list=PLe6EXFvnTV7-_41SpakZoTIYCgX4aMTdU&index=2&t=0s) and share your changes.
 If you're new to git, we recommend [GitKraken](https://www.gitkraken.com/) for an intuitive interface.
 
-Please submit any direct contributions by [forking the repository](https://www.youtube.com/watch?v=Lb4yvfrX_7I&list=PLe6EXFvnTV7-_41SpakZoTIYCgX4aMTdU&index=3&t=9s), creating a branch, and [submitting a Pull Request](https://www.youtube.com/watch?v=2VX1ISk9XH8&list=PLe6EXFvnTV7-_41SpakZoTIYCgX4aMTdU&index=9&t=0s).
+We have several "rules" for contributions:
+
+1. If your contribution is complex, adds new features, new UI design or otherwise warrants discussion, we highly recommend that you submit a github issue, and engage other contributors in a discussion about the details of your proposed PR. This will save you time in the long run, as many details and decisions can be hashed out ahead-of-time.
+
+2. Please submit any direct contributions by [forking the repository](https://www.youtube.com/watch?v=Lb4yvfrX_7I&list=PLe6EXFvnTV7-_41SpakZoTIYCgX4aMTdU&index=3&t=9s), creating a feature branch, and [submitting a Pull Request](https://www.youtube.com/watch?v=2VX1ISk9XH8&list=PLe6EXFvnTV7-_41SpakZoTIYCgX4aMTdU&index=9&t=0s). **Do not submit pull requests against master.**
```

Comment:
> **Do not submit pull requests against master.**

Not sure I understand this bit. If they create a fork and submit a pull request against cellxgene, it _would_ be a PR against `cellxgene:master`.

Re. the documentation videos: why not just use Github's documentation on these topics ([creating a PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request), [submitting a PR from a fork](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork))? It's specifically tailored to Github and doesn't introduce any extra services like GitKraken.

Reply:
Documentation: good idea, will fix. I'm not sure why we ever made such recommendations or used something other than official docs.  GitKraken has known issues (even our team has trouble with it), so I'll excise the recommendation.

Side note: various other projects, such as numpy, have great summary workflow.  More of a [recipe](https://docs.scipy.org/doc/numpy/dev/).  Thoughts on referencing those as examples?  I actually find our instructions confusing as they skip the fork.

On my badly worded edits -- the concept I'm trying to communicate is to always create a feature branch -- don't modify master on your fork.

Reply:
> Side note: various other projects, such as numpy, have great summary workflow. More of a recipe. Thoughts on referencing those as examples? I actually find our instructions confusing as they skip the fork.

If anything, we should probably have our own recipe, complete with links out to how users can test any of the changes they make, run our linters/formatters, and create documentation. That seems to be the motivation for why numpy guide exists. Otherwise, they could simply point to something like the GitHub docs.

Reply:
yep, and for now, I took your suggestion and pointed at github docs.  The problem with those docs is that they are a big lift compared to something like nice like [the numpy contrib recipe](https://docs.scipy.org/doc/numpy/dev/).